ios/Classes/Downloader.swift

```diff
@@ -384,7 +384,12 @@ public class Downloader: NSObject, FlutterPlugin, URLSessionDelegate, URLSession
             result(nil)
             return
         }
-        result(moveToSharedStorage(filePath: filePath, destination: destination, directory: directory))
+        moveToSharedStorage(
```

Comment:
To make this simpler and more consistent with how we've handled the result, I suggested adding a parameter `result: @escaping FlutterResult` to the new `moveToSharedStorage` function. From there you can then simply call `result(yourPath)` or `result(nil)` to pass the result back to Flutter.  You do then need to follow that statement with `return` to exit the method.
What you have done here is adding another `completer` layer to the process, which makes the code a bit hard to read/follow

Reply:
Yeah.  Several methods need to use callbacks here.

I was trying to prevent the FlutterResult from leaking to peripheral modules, but that's easy to change.